tests/simulator/py/em_init.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+#
+# formatted with black
+
+from tests.simulator.py import InitValueValidation
+
+import numpy as np
+
+
+class EMInitValidation(InitValueValidation):
+    is_primal = {"bx": 1, "by": 0, "bz": 0, "ex": 0, "ey": 1, "ez": 1}
+
+    def test_1d(self):
+        from phare.pp.diagnostics import _EM
+        from tests.diagnostic import dump_all_diags
+
+        diag_out_dir = "phare_outputs/em_init"
+        dic = InitValueValidation.diag_options(diag_out_dir)
+        dic.update({"diags_fn": lambda model: dump_all_diags(model.populations)})
+        diags = self._simulate_diagnostics(dim=1, interp=1, input=dic)
+        self._checkEMFromFN(diags[_EM.__name__])
+
+    def _checkEMFromFN(self, ems):
+        truncation, tolerance = 5, 1e-4  # the nearness is not great
```

Comment:
it's not

Reply:
that is suspicious. It should be, and if not we need to understand why

Reply:
this might be something we should look further into

if we have discrepancies between 
A) static_cast<float>(double{1.11...})
B) np.array(array, dtype=np.float32)

Reply:
I don't understand your comment

Reply:
as to why there is a difference between the arrays

In C++ we cast all datasets to float when dumping

Reply:
yes ok we dump in float. So we need to compare with user function giving float values. Do we have equality between the two at the float precision?

Reply:
to five decimal places at least, mostly more